src/actions/get-withdrawal-data.ts

```diff
@@ -0,0 +1,73 @@
+"use server";
+import { env } from "@/env";
+
+import { BufferCV, Cl, TupleCV, UIntCV } from "@stacks/transactions";
+
+import { encodeBitcoinAddress } from "@/util/decode-bitcoin-address";
+import { WithdrawalStatus } from "@/app/withdraw/[txid]/components/util";
+import getBitcoinNetwork from "@/util/get-bitcoin-network";
+
+type contractCallData = {
+  contract_id: string;
+  function_name: string;
+  function_signature: string;
+  function_args: FunctionArg[];
+};
+
+type FunctionArg = {
+  hex: string;
+  repr: string;
+  name: string;
+  type: string;
+};
+
+export async function getWithdrawalInfo(txid: string) {
+  const tx = await fetch(`${env.STACKS_API_URL}/extended/v1/tx/${txid}`);
+  const txData = await tx.json();
+  const contractCall = txData.contract_call as contractCallData;
```

Comment:
Should we handle the not found case explicitly? As in, if the tx is not yet available in the stack api we are querying.

Also below for emily returning a non success.

Reply:
the tx not yet available in the stacks api is not possible since we broadcast the tx ourselves, if emily returns a non success it will be retried

Reply:
Just to confirm: if emily returns an error `withdrawal.json()` will throw, or if it's a json encoded error the returned `status` may be undefined. Is this handled correctly?

Reply:
no good catch need to handle that correctly however if the request returns ok status I'm assuming must exist

Reply:
should be fixed now